test/boost/secondary_index_test.cc

```diff
@@ -1518,6 +1518,37 @@ SEASTAR_TEST_CASE(test_computed_columns) {
     });
 }
 
+// Ref: #3423 - rows should be returned in token order,
+// using signed comparison (ref: #7443)
+SEASTAR_TEST_CASE(test_token_order) {
```

Comment:
As a sanity check, please make sure this test *fails* before your patch :-)

Reply:
Tested it: only adding this new test without any other changes - it fails.